### ISSUE TITLE
[Security] Vulnerability in `tar`

### ISSUE THREAD

**Issue by jeanpylone** [gruntjs/grunt-contrib-compress#210]
See https://www.npmjs.com/advisories/803

> Versions of node-tar prior to 4.4.2 are vulnerable to Arbitrary File Overwrite. Extracting tarballs containing a hardlink to a file that already exists in the system, and a file that matches the hardlink will overwrite the system's file with the contents of the extracted file.

This `tar` dependency comes from `iltorb` dependency. Latest version of `iltorb` seem to correct the issue.

```
~$ npm audit
                                                                                
                       === npm audit security report ===                        
                                                                                
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary File Overwrite                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ tar                                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.4.2                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ grunt-contrib-compress [dev]                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ grunt-contrib-compress > iltorb > node-gyp > tar             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/803                             │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by jakeIdFyed** [comment]
Any update on this? I'm still facing the same issue.

**Comment by vladikoff** [comment]
Try version 1.5.0!

**Comment by jeanpylone** [comment]
I fear you can reopen this issue.

``` sh
+ grunt-contrib-compress@1.5.0
added 155 packages from 120 contributors and audited 477 packages in 67.653s
found 1 high severity vulnerability
  run 'npm audit fix' to fix them, or 'npm audit' for details

jp@redxiii ~/Documents/projects/tests/compress $ npm audit
                                                                                
                       === npm audit security report ===                        
                                                                                
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary File Overwrite                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ tar                                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.4.2                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ grunt-contrib-compress [dev]                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ grunt-contrib-compress > iltorb > node-gyp > tar             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/803                             │
└───────────────┴──────────────────────────────────────────────────────────────┘
found 1 high severity vulnerability in 477 scanned packages
  1 vulnerability requires manual review. See the full report for details.

jp@redxiii ~/Documents/projects/tests/compress $ npm list
compress@1.0.0 /home/jp/Documents/projects/tests/compress
└─┬ grunt-contrib-compress@1.5.0
  ├─┬ archiver@1.3.0
  │ ├─┬ archiver-utils@1.3.0
  │ │ ├── glob@7.1.3 deduped
  │ │ ├── graceful-fs@4.1.15
  │ │ ├─┬ lazystream@1.0.0
  │ │ │ └── readable-stream@2.3.6 deduped
  │ │ ├── lodash@4.17.11 deduped
  │ │ ├─┬ normalize-path@2.1.1
  │ │ │ └── remove-trailing-separator@1.1.0
  │ │ └── readable-stream@2.3.6 deduped
  │ ├─┬ async@2.6.2
  │ │ └── lodash@4.17.11 deduped
  │ ├── buffer-crc32@0.2.13
  │ ├─┬ glob@7.1.3
  │ │ ├── fs.realpath@1.0.0
  │ │ ├─┬ inflight@1.0.6
  │ │ │ ├── once@1.4.0 deduped
  │ │ │ └── wrappy@1.0.2
  │ │ ├── inherits@2.0.3
  │ │ ├─┬ minimatch@3.0.4
  │ │ │ └─┬ brace-expansion@1.1.11
  │ │ │   ├── balanced-match@1.0.0
  │ │ │   └── concat-map@0.0.1
  │ │ ├─┬ once@1.4.0
  │ │ │ └── wrappy@1.0.2 deduped
  │ │ └── path-is-absolute@1.0.1
  │ ├── lodash@4.17.11 deduped
  │ ├─┬ readable-stream@2.3.6
  │ │ ├── core-util-is@1.0.2
  │ │ ├── inherits@2.0.3 deduped
  │ │ ├── isarray@1.0.0
  │ │ ├── process-nextick-args@2.0.0
  │ │ ├── safe-buffer@5.1.2
  │ │ ├─┬ string_decoder@1.1.1
  │ │ │ └── safe-buffer@5.1.2 deduped
  │ │ └── util-deprecate@1.0.2
  │ ├─┬ tar-stream@1.6.2
  │ │ ├─┬ bl@1.2.2
  │ │ │ ├── readable-stream@2.3.6 deduped
  │ │ │ └── safe-buffer@5.1.2 deduped
  │ │ ├─┬ buffer-alloc@1.2.0
  │ │ │ ├── buffer-alloc-unsafe@1.1.0
  │ │ │ └── buffer-fill@1.0.0
  │ │ ├─┬ end-of-stream@1.4.1
  │ │ │ └── once@1.4.0 deduped
  │ │ ├── fs-constants@1.0.0
  │ │ ├── readable-stream@2.3.6 deduped
  │ │ ├── to-buffer@1.1.1
  │ │ └── xtend@4.0.1
  │ ├── walkdir@0.0.11
  │ └─┬ zip-stream@1.2.0
  │   ├── archiver-utils@1.3.0 deduped
  │   ├─┬ compress-commons@1.2.2
  │   │ ├── buffer-crc32@0.2.13 deduped
  │   │ ├─┬ crc32-stream@2.0.0
  │   │ │ ├─┬ crc@3.8.0
  │   │ │ │ └─┬ buffer@5.2.1
  │   │ │ │   ├── base64-js@1.3.0
  │   │ │ │   └── ieee754@1.1.13
  │   │ │ └── readable-stream@2.3.6 deduped
  │   │ ├── normalize-path@2.1.1 deduped
  │   │ └── readable-stream@2.3.6 deduped
  │   ├── lodash@4.17.11 deduped
  │   └── readable-stream@2.3.6 deduped
  ├─┬ chalk@1.1.3
  │ ├── ansi-styles@2.2.1
  │ ├── escape-string-regexp@1.0.5
  │ ├─┬ has-ansi@2.0.0
  │ │ └── ansi-regex@2.1.1
  │ ├─┬ strip-ansi@3.0.1
  │ │ └── ansi-regex@2.1.1 deduped
  │ └── supports-color@2.0.0
  ├─┬ iltorb@1.3.10
  │ ├── detect-libc@0.2.0
  │ ├── nan@2.13.2
  │ ├─┬ node-gyp@3.8.0
  │ │ ├─┬ fstream@1.0.11
  │ │ │ ├── graceful-fs@4.1.15 deduped
  │ │ │ ├── inherits@2.0.3 deduped
  │ │ │ ├── mkdirp@0.5.1 deduped
  │ │ │ └── rimraf@2.6.3 deduped
  │ │ ├── glob@7.1.3 deduped
  │ │ ├── graceful-fs@4.1.15 deduped
  │ │ ├─┬ mkdirp@0.5.1
  │ │ │ └── minimist@0.0.8
  │ │ ├─┬ nopt@3.0.6
  │ │ │ └── abbrev@1.1.1
  │ │ ├─┬ npmlog@4.1.2
  │ │ │ ├─┬ are-we-there-yet@1.1.5
  │ │ │ │ ├── delegates@1.0.0
  │ │ │ │ └── readable-stream@2.3.6 deduped
  │ │ │ ├── console-control-strings@1.1.0
  │ │ │ ├─┬ gauge@2.7.4
  │ │ │ │ ├── aproba@1.2.0
  │ │ │ │ ├── console-control-strings@1.1.0 deduped
  │ │ │ │ ├── has-unicode@2.0.1
  │ │ │ │ ├── object-assign@4.1.1
  │ │ │ │ ├── signal-exit@3.0.2
  │ │ │ │ ├─┬ string-width@1.0.2
  │ │ │ │ │ ├── code-point-at@1.1.0
  │ │ │ │ │ ├─┬ is-fullwidth-code-point@1.0.0
  │ │ │ │ │ │ └── number-is-nan@1.0.1
  │ │ │ │ │ └── strip-ansi@3.0.1 deduped
  │ │ │ │ ├── strip-ansi@3.0.1 deduped
  │ │ │ │ └─┬ wide-align@1.1.3
  │ │ │ │   └── string-width@1.0.2 deduped
  │ │ │ └── set-blocking@2.0.0
  │ │ ├─┬ osenv@0.1.5
  │ │ │ ├── os-homedir@1.0.2 deduped
  │ │ │ └── os-tmpdir@1.0.2
  │ │ ├─┬ request@2.88.0
  │ │ │ ├── aws-sign2@0.7.0
  │ │ │ ├── aws4@1.8.0
  │ │ │ ├── caseless@0.12.0
  │ │ │ ├─┬ combined-stream@1.0.7
  │ │ │ │ └── delayed-stream@1.0.0
  │ │ │ ├── extend@3.0.2
  │ │ │ ├── forever-agent@0.6.1
  │ │ │ ├─┬ form-data@2.3.3
  │ │ │ │ ├── asynckit@0.4.0
  │ │ │ │ ├── combined-stream@1.0.7 deduped
  │ │ │ │ └── mime-types@2.1.24 deduped
  │ │ │ ├─┬ har-validator@5.1.3
  │ │ │ │ ├─┬ ajv@6.10.0
  │ │ │ │ │ ├── fast-deep-equal@2.0.1
  │ │ │ │ │ ├── fast-json-stable-stringify@2.0.0
  │ │ │ │ │ ├── json-schema-traverse@0.4.1
  │ │ │ │ │ └─┬ uri-js@4.2.2
  │ │ │ │ │   └── punycode@2.1.1
  │ │ │ │ └── har-schema@2.0.0
  │ │ │ ├─┬ http-signature@1.2.0
  │ │ │ │ ├── assert-plus@1.0.0
  │ │ │ │ ├─┬ jsprim@1.4.1
  │ │ │ │ │ ├── assert-plus@1.0.0 deduped
  │ │ │ │ │ ├── extsprintf@1.3.0
  │ │ │ │ │ ├── json-schema@0.2.3
  │ │ │ │ │ └─┬ verror@1.10.0
  │ │ │ │ │   ├── assert-plus@1.0.0 deduped
  │ │ │ │ │   ├── core-util-is@1.0.2 deduped
  │ │ │ │ │   └── extsprintf@1.3.0 deduped
  │ │ │ │ └─┬ sshpk@1.16.1
  │ │ │ │   ├─┬ asn1@0.2.4
  │ │ │ │   │ └── safer-buffer@2.1.2 deduped
  │ │ │ │   ├── assert-plus@1.0.0 deduped
  │ │ │ │   ├─┬ bcrypt-pbkdf@1.0.2
  │ │ │ │   │ └── tweetnacl@0.14.5 deduped
  │ │ │ │   ├─┬ dashdash@1.14.1
  │ │ │ │   │ └── assert-plus@1.0.0 deduped
  │ │ │ │   ├─┬ ecc-jsbn@0.1.2
  │ │ │ │   │ ├── jsbn@0.1.1 deduped
  │ │ │ │   │ └── safer-buffer@2.1.2 deduped
  │ │ │ │   ├─┬ getpass@0.1.7
  │ │ │ │   │ └── assert-plus@1.0.0 deduped
  │ │ │ │   ├── jsbn@0.1.1
  │ │ │ │   ├── safer-buffer@2.1.2
  │ │ │ │   └── tweetnacl@0.14.5
  │ │ │ ├── is-typedarray@1.0.0
  │ │ │ ├── isstream@0.1.2
  │ │ │ ├── json-stringify-safe@5.0.1
  │ │ │ ├─┬ mime-types@2.1.24
  │ │ │ │ └── mime-db@1.40.0
  │ │ │ ├── oauth-sign@0.9.0
  │ │ │ ├── performance-now@2.1.0
  │ │ │ ├── qs@6.5.2
  │ │ │ ├── safe-buffer@5.1.2 deduped
  │ │ │ ├─┬ tough-cookie@2.4.3
  │ │ │ │ ├── psl@1.1.31
  │ │ │ │ └── punycode@1.4.1
  │ │ │ ├── tunnel-agent@0.6.0 deduped
  │ │ │ └── uuid@3.3.2
  │ │ ├─┬ rimraf@2.6.3
  │ │ │ └── glob@7.1.3 deduped
  │ │ ├── semver@5.3.0
  │ │ ├─┬ tar@2.2.1
  │ │ │ ├─┬ block-stream@0.0.9
  │ │ │ │ └── inherits@2.0.3 deduped
  │ │ │ ├── fstream@1.0.11 deduped
  │ │ │ └── inherits@2.0.3 deduped
  │ │ └─┬ which@1.3.1
  │ │   └── isexe@2.0.0
  │ └─┬ prebuild-install@2.5.3
  │   ├── detect-libc@1.0.3
  │   ├── expand-template@1.1.1
  │   ├── github-from-package@0.0.0
  │   ├── minimist@1.2.0
  │   ├── mkdirp@0.5.1 deduped
  │   ├─┬ node-abi@2.7.1
  │   │ └── semver@5.7.0
  │   ├── noop-logger@0.1.1
  │   ├── npmlog@4.1.2 deduped
  │   ├── os-homedir@1.0.2
  │   ├─┬ pump@2.0.1
  │   │ ├── end-of-stream@1.4.1 deduped
  │   │ └── once@1.4.0 deduped
  │   ├─┬ rc@1.2.8
  │   │ ├── deep-extend@0.6.0
  │   │ ├── ini@1.3.5
  │   │ ├── minimist@1.2.0
  │   │ └── strip-json-comments@2.0.1
  │   ├─┬ simple-get@2.8.1
  │   │ ├─┬ decompress-response@3.3.0
  │   │ │ └── mimic-response@1.0.1
  │   │ ├── once@1.4.0 deduped
  │   │ └── simple-concat@1.0.0
  │   ├─┬ tar-fs@1.16.3
  │   │ ├── chownr@1.1.1
  │   │ ├── mkdirp@0.5.1 deduped
  │   │ ├─┬ pump@1.0.3
  │   │ │ ├── end-of-stream@1.4.1 deduped
  │   │ │ └── once@1.4.0 deduped
  │   │ └── tar-stream@1.6.2 deduped
  │   ├─┬ tunnel-agent@0.6.0
  │   │ └── safe-buffer@5.1.2 deduped
  │   └── which-pm-runs@1.0.0
  ├── lodash@4.17.11
  ├── pretty-bytes@4.0.2
  └── stream-buffers@2.2.0


```

**Comment by fwextensions** [comment]
#209 incremented the minor version of `iltorb`, but that package is still including `tar ^2.0.0` via `node-gyp`, so npm is still complaining.  The latest major version of `iltorb` is fine.
